### ISSUE TITLE
Completion on `out` in local function impedes with typing

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#22253]
I tried to add an out parameter to a local function and Intellisense forced a completion of a type name instead of `out`.

![image](https://user-images.githubusercontent.com/12466233/30706186-79bb6e1e-9eac-11e7-8d4a-d80452857185.png)

FYI @agocke @Pilchie for triage

**Comment by dpoeschl** [comment]
The `OutKeywordRecommender` doesn't seem to be kicking in here. Can you take a look?

**Comment by dpoeschl** [comment]
Smaller standalone repro:

```C#
using System;

class C
{
    void M()
    {
        void F(int x, $$)
    }
}
```

**Comment by chborl** [comment]
Issue fixed by #17447 which was merged into master with #22050.  
Tests added in #22388
